include/http/request_handler.js

```diff
@@ -534,15 +535,26 @@ module.exports = function RequestHandlerModule(pb) {
      */
     RequestHandler.prototype.handleRequest = function(){
 
-        //fist things first check for public resource
+        //first things first check for public resource
         if (RequestHandler.isPublicRoute(this.url.pathname)) {
             return this.servePublicContent();
         }
 
         //check for session cookie
         var cookies = RequestHandler.parseCookies(this.req);
+        var siteObj = RequestHandler.sites[this.hostname];
+        var url = this.url.pathname;
         this.req.headers[pb.SessionHandler.COOKIE_HEADER] = cookies;
 
+        if(typeof(siteObj) !== 'undefined' && siteObj.hasOwnProperty('forceLocale') && siteObj.forceLocale) {
+            cookies.locale = cookies.locale || cookies[" locale"];
+            cookies.locale = cookies.locale || siteObj.defaultLocale;
+            if (checkRouteContainsLocale(url, siteObj)) {
+                var redirectLocation = pb.SiteService.getHostWithProtocol(this.hostname);
+                redirectLocation += '/' + cookies.locale + this.url.path;
+                return this.doRedirect(redirectLocation, pb.HttpStatus.MOVED_TEMPORARILY);
+            }
+        }
```

Comment:
I don't think i have a good handle on what the angular plugin is supposed to do?  Is it forcing a specific locale other than what is being sent in the locale header? We parse the locale header each time it is sent to the server (seems to be exactly what the cookie is doing).   

We won't accept this PR until we get a better understanding of why this is needed in the platform and can ensure that all pieces of the puzzle cookie creation and value extraction are part of the platform.  It doesn't make sense for us to put the other half of a plugin into the platform.

Is the real issue that you can't hook into the request handler from a plugin?